### PR TITLE
Add support for collapsible callouts

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2348** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2348" target="_blank">Graphite</a> 
    * **PR #2349** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2349" target="_blank">Graphite</a> 
      * **PR #2350** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2350" target="_blank">Graphite</a> 
        * **PR #2351** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2351" target="_blank">Graphite</a> 
          * **PR #2368** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2368" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2368?utm_source=stack-comment).

File: src/components/Callout.tsx [hunk 1]
```diff
@@ -1,37 +1,85 @@
...
         display: 'flex',
         alignItems: 'flex-start',
+        gap: '1rem',
+
+        cursor: collapsible ? 'pointer' : undefined,
+        userSelect: collapsible ? 'none' : undefined,
+
         ...style,
       }}
+      onClick={collapsible ? handleToggleExpand : undefined}
```
Comment (tomquirk) [inline src/components/Callout.tsx:60]: **suggestion** add a sneaky `role="button"` for good measure? :)

File: src/components/Create/components/pages/ProjectToken/components/CustomTokenSettings/components/ReservedTokenRateCallout/ReservedTokenRateCallout.tsx [hunk 2]
```diff
@@ -25,12 +25,20 @@ export const ReservedTokenRateCallout: React.FC = () => {
...
-    <Callout>
-      <Trans>
-        Contributor rate: {formattedNum(contributorTokens)} / 1 ETH
-        <br />
-        Reserved rate: {formattedNum(reservedTokens)} / 1 ETH
-      </Trans>
-    </Callout>
+    <CreateCallout.Info noIcon collapsible={false}>
+      <span>
+        <Trans>Contributor rate:</Trans>
```
Comment (tomquirk) [inline src/components/Create/components/pages/ProjectToken/components/CustomTokenSettings/components/ReservedTokenRateCallout/ReservedTokenRateCallout.tsx:30]: **suggestion** still good to have the `Trans` encapsulate the full `key: value`. In some languages it reads `value: key`.

File: src/components/Create/components/pages/ProjectToken/components/CustomTokenSettings/CustomTokenSettings.tsx [hunk 3]
```diff
@@ -136,12 +136,10 @@ export const CustomTokenSettings = () => {
...
           <CreateCallout.Warning>
-            <Space direction="vertical">
-              <Trans>
-                The discount rate cannot be set while the funding cycle duration
-                is not set.
-              </Trans>
-            </Space>
+            <Trans>
+              The discount rate cannot be set while the funding cycle duration
+              is not set.
```
Comment (tomquirk) [inline src/components/Create/components/pages/ProjectToken/components/CustomTokenSettings/CustomTokenSettings.tsx:141]:
```suggestion
              The discount rate cannot be set when funding cycle duration
              is not set.
```

**Comment by tomquirk** [comment]
@wraeth-eth mind adding a screenshot or video demo for both states of the Callout?

**Comment by wraeth-eth** [comment]
https://user-images.githubusercontent.com/104132113/198166123-20a7a99c-5a90-4a6e-b6be-58c8c29a1877.mov

@aeolianeth

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2368?utm_source=stack-comment).


Job Reference: tq0DBErRQ758wBw1xKNP

**Pull request merged**